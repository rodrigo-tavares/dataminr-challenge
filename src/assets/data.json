{
  "features": [
    {
      "name": "General",
      "items": [
        {
          "title": "Case management",
          "value": false
        },
        {
          "title": "Map timeline",
          "value": false
        },
        {
          "title": "Views & Briefings",
          "value": false
        },
        {
          "title": "Notifications",
          "value": false
        },
        {
          "title": "Mass communications",
          "value": false
        },
        {
          "title": "Traffic Cameras",
          "value": false
        }
      ]
    },
    {
      "name": "Settings",
      "childs": [
        {
          "title": "Audit log",
          "value": false
        },
        {
          "title": "Users",
          "value": false,
          "children": [
            {
              "title": "Users Add",
              "value": false
            },
            {
              "title": "User Delete",
              "value": false
            },
            {
              "title": "Users Edit",
              "value": false
            },
            {
              "title": "Max users",
              "value": false,
              "maxValue": 100,
              "initialValue": 0,
              "selectedValue": 30
            }
          ]
        }
      ]
    },
    {
      "name": "Alerts",
      "childs": [
        {
          "title": "Alert manager",
          "value": false
        },
        {
            "title": "Alert Rules",
            "value": false,
            "maxValue": 100,
            "initialValue": 0,
            "selectedValue": 30
          }
      ]
    }
  ]
}
