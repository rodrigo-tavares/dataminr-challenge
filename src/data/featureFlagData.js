import jsonData from './../assets/data.json';

export const getFormData = () => jsonData;