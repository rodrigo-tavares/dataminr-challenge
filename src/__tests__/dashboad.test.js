import { mount } from "enzyme";
import React from "react";
import Dashboard from "../components/dashboard";
import renderer from "react-test-renderer";

// eslint-disable-next-line jest/no-mocks-import
import { dataMock } from "../__mocks__/data.mock";

jest.mock("../__mocks__/data.mock.js");

const props = {
  features: dataMock.features,
};

describe("Dashboard component", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly", () => {
    const tree = renderer.create(<Dashboard {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should receive at least the mock features", () => {
    const customProps = {
      features: dataMock.features,
    }, view = mount(<Dashboard {...customProps} />);


    expect((view).prop('features')).toEqual(dataMock.features);
  });

  it("should receive a empty array of feature", () => {
    const customProps = {
      features: [],
    }, view = mount(<Dashboard {...customProps} />);


    expect((view).prop('features')).toEqual([]);
  });
});
