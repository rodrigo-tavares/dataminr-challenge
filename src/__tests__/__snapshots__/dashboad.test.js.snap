// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component renders correctly 1`] = `
<div
  className="sc-bqiRlB haKyjn"
>
  <div
    className="sc-bdvvtL dXbRqN"
  >
    No data provided!
  </div>
</div>
`;
